app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -0,0 +1,81 @@
+module Katello
+  class Api::V2::HostSubscriptionsController < Katello::Api::V2::ApiController
+    respond_to :json
+
+    before_filter :find_host_with_subscriptions
+
+    api :GET, "/hosts/:id/subscriptions", N_("List a host's subscriptions"), :deprecated => true
+    param :id, String, :desc => N_("id of the host"), :required => true
+    def index
+      @collection = index_response
+      respond_for_index :collection => @collection
+    end
+
+    def index_response
+      entitlements = @candlepin_consumer.entitlements
+      subscriptions =  entitlements.map { |entitlement| ::Katello::SystemSubscriptionPresenter.new(entitlement) }
+      full_result_response(subscriptions)
+    end
+
+    api :PUT, "/hosts/:id/subscriptions/auto_attach", N_("Trigger an auto-attach of subscriptions")
+    param :id, String, :desc => N_("id of the host"), :required => true
+    def auto_attach
+      sync_task(::Actions::Katello::Host::AutoAttachSubscriptions, @host)
+      respond_for_index(:collection => index_response, :template => "index")
+    end
+
+    api :GET, "/hosts/:id/subscriptions/events", N_("List Candlepin events for the content host")
+    param :id, String, :desc => N_("UUID of the content host"), :required => true
+    def events
+      respond_for_index :collection => { :results => @candlepin_consumer.events.map { |e| OpenStruct.new(e) } }
+    end
+
+    api :PUT, "/hosts/:host_id/subscriptions/remove_subscriptions"
+    param :subscriptions, Array, :desc => N_("Array of subscriptions to add"), :required => false do
+      param :id, String, :desc => N_("Subscription Pool uuid")
+      param :quantity, Integer, :desc => N_("If specified, remove the first instance of a subscription with matching id and quantity.")
+    end
+    def remove_subscriptions
+      #combine the quantities for duplicate pools into PoolWithQuantities objects
+      pool_id_quantities = params[:subscriptions].inject({}) do |new_hash, subscription|
+        new_hash[subscription['id']] ||= PoolWithQuantities.new(Pool.find(subscription['id']))
+        new_hash[subscription['id']].quantities << subscription['quantity']
+        new_hash
+      end
+
+      @host.subscription_facet.remove_subscriptions(pool_id_quantities.values)
+      respond_for_index(:collection => index_response, :template => "index")
+    end
+
+    api :PUT, "/hosts/:host_id/subscriptions/add_subscriptions", N_("Add a subscription to a host"), :deprecated => true
```

Comment:
I thought sub resources typically use :top_level_resource_id and not :id for the top level resources.  Or am i wrong?

In REST isn't  POST /hosts/:id/subscriptions generally used for creating a subscription?  Not associating it to that subscription.  Idealy this would be done with a put to /hosts/:id  as we are ally updating the host, but subscriptions are harder to do that with.